### DISCUSSION TITLE
Procedural content generation

### DISCUSSION THREAD

**Discussion by pierrethibault** [executablebooks/meta#517] [Q&A]
Is there a way to use a jupyter notebook to generate MyST content? An example use case would be to loop through images to display, or through elements in a dictionary to create an index.

**Comment by choldgraf** [comment]
You should be able to display markdown that will be parsed as CommonMark. See this section for some details: http://127.0.0.1:5501/docs/_build/html/content/code-outputs.html#markdown

**Reply by pierrethibault** [reply]
Thanks. I had seen this already, but it's not clear to me that I can do everything that MyST offers. I don't see how to get numbered figures or other MyST directives for instance.

If there is no way to do it inside a notebook I suppose the simplest is to generate a complete file that's already inserted in the toc?

**Reply by pierrethibault** [reply]
Here's another use case: I am about to write a long book and I don't feel like committing to a fixed figure size before I see the final result. I haven't found a way to use a variable for directive parameters. One way would be to "wrap" a directive into a python function, but that won't work if there is no way to inject it back in the document.
